src/main/scala/dev/kamu/cli/commands/PullCommand.scala

```diff
@@ -20,15 +20,44 @@ class PullCommand(
   private val logger = LogManager.getLogger(getClass.getName)
 
   def run(): Unit = {
-    val datasets =
-      if (all)
-        metadataRepository.getAllDatasets()
-      else
-        datasetIDs
-          .map(DatasetID)
-          .map(metadataRepository.getDataset)
-
-    val numUpdated = datasets
+    val s = metadataRepository.getAllDatasetIDs()
+    val datasets: IndexedSeq[DatasetID] = s.toIndexedSeq
+    val datasetIntID = s.zipWithIndex.toMap
+    val idToInt: DatasetID => Int = d => datasetIntID(d)
+    val intToId: Int => DatasetID = d => datasets(d)
+    val idToDataset: DatasetID => Dataset = d => metadataRepository.getDataset(d)
```

Comment:
Please install Scalafmt plugin for IntelliJ and configure it to format code each time you save a file to make sure our code is consistently formatted.